posydon/popsyn/synthetic_population.py

```diff
@@ -992,16 +992,31 @@ def __init__(
                 Pwarn(f"{filename} already contains a mass_per_metallicity "
                       "table. Overwriting the table!", "OverwriteWarning")
 
-            simulated_mass = np.sum(self.oneline[["S1_mass_i", "S2_mass_i"]].to_numpy())
+            if "initially_single_star" in self.oneline["state_i"].values:
+                mask=self.oneline["state_i"]=="initially_single_star"
+                filtered_data_single = self.oneline[mask]
+                filtered_data_binaries = self.oneline[~mask]
+    
+                simulated_mass_binaries = np.sum(filtered_data_binaries[["S1_mass_i", "S2_mass_i"]].to_numpy())
+                simulated_mass_single = np.sum(filtered_data_single[["S1_mass_i"]].to_numpy())
+                simulated_mass = simulated_mass_binaries + simulated_mass_single
+            else:
+                simulated_mass_binaries = np.sum(self.oneline[["S1_mass_i", "S2_mass_i"]].to_numpy())
+                simulated_mass_single = None
```

Comment:
May just a style issue, but wouldn't it be saver to `set simulated_mass_single = 0.0` here, to ensure that it is a float and not a none type object. Usually, we use None for undefined objects, but here I'd say the simulated mass of single stars is well defined to be zero, because they didn't get simulated.